### DISCUSSION TITLE
Shell Integration Contribute?

### DISCUSSION THREAD

**Discussion by JustinGrote** [microsoft/vscode-discussions#933] [Extension Development QnA] [answered]
Is it possible to contribute items to either the shell integration status menu or the menu that appears when you click it? In the PowerShell extension we control the execution so we want to add additional info such as command timing information or Feedback Provider suggestions.

![image](https://github.com/microsoft/vscode-discussions/assets/15258962/3a068824-1849-45b4-85f8-db654da24d3c)
![image](https://github.com/microsoft/vscode-discussions/assets/15258962/cadbd0cc-ffac-42ee-b532-fe00cda7ce99)

**Comment by alefragnani** [comment, marked as answer]
Hi @JustinGrote ,

This context menu doesn't seem to be exposed for contributions, as for terminals, you only have `terminal/context` and `terminal/title/context`. But, at the same time, the [Contribution Points](https://code.visualstudio.com/api/references/contribution-points#contributes.submenus) documentation says that _any contributes submenu_ could receive new items. The point is, what's the name of that context menu?

Also, maybe [this API proposal](https://github.com/microsoft/vscode/issues/145234) may be a good start. It talks about _Expose shell integration command knowledge to extensions_, but if extensions can know about the shell integration, why not also contribute on that context menu or that tooltip, to provide additional info/commands.

Hope this helps

**Reply by JustinGrote** [reply]
Thanks @alefragnani, https://github.com/microsoft/vscode/issues/162950 is exactly what I was looking for.
